r/tests/testthat/test-Array.R

```diff
@@ -65,230 +75,136 @@ test_that("Array supports NA", {
   expect_true(x_int$IsNull(10L))
   expect_true(x_dbl$IsNull(10L))
```

Comment:
That's originally what I was planning to do in this branch, fill in coverage for those methods (and make sure that `x_int$IsNull("ten")` doesn't segfault), but now I think I'll take that up separately since this PR is big enough already and it fixes a different bug.